InterviewQuestionsTests/Chapters/ArraysStrings/ArraysStrings.Tests.cs

```diff
@@ -19,37 +19,45 @@ public void isStringUniqTest(string input, bool expectedResult)
             bool output = testObject.isStringUniq(input);
             bool outputSecond = testObject.isStringUniqSecondSolution(input);
 
-            
+
             Assert.AreEqual(expectedResult, output);
-            Assert.AreEqual(expectedResult, outputSecond);    
+            Assert.AreEqual(expectedResult, outputSecond);
         }
-        
+
         [DataTestMethod]
-        [DataRow("aaabbcccccdd","a3b2c5d2")]
+        [DataRow("aaabbcccccdd", "a3b2c5d2")]
         [DataRow("aabcde", "aabcde")]
         [DataRow("aaaaabccdeee", "a5b1c2d1e3")]
-        [DataRow("abcdef","abcdef" )]
+        [DataRow("abcdef", "abcdef")]
         public void stringCompression(string input, string expectedResult)
         {
             ArraysAndStringSolutions testObject1 = new ArraysAndStringSolutions();
 
             string solution = testObject1.stringCompression(input);
-            string alternateSolution =testObject1.stringCompressionAlternate(input);
-            
+            string alternateSolution = testObject1.stringCompressionAlternate(input);
+
             Assert.AreEqual(expectedResult, solution);
-            Assert.AreEqual(expectedResult,alternateSolution);
+            Assert.AreEqual(expectedResult, alternateSolution);
         }
         [DataTestMethod]
-        [DataRow("abdulsamed", "demasludba",true)]
-        [DataRow("furkan", "furcan",false)]
-        [DataRow("ali", "lia",true)]
-        public void stringPermutation(string input,string testString,bool expectedResult)
+        [DataRow("abdulsamed", "demasludba", true)]
+        [DataRow("furkan", "furcan", false)]
+        [DataRow("ali", "lia", true)]
+        public void stringPermutation(string input, string testString, bool expectedResult)
         {
             ArraysAndStringSolutions testObjecet2 = new ArraysAndStringSolutions();
 
-            bool solution = testObjecet2.isStringPermutation(input,testString);
+            bool solution = testObjecet2.isStringPermutation(input, testString);
+
+            Assert.AreEqual(expectedResult, solution);
+        }
+        public void zeroMatrix(int[,] input, int[,] expectedResult)
+        {
+            ArraysAndStringSolutions testObject = new ArraysAndStringSolutions();
+
+            int[,] solution = testObject.zeroesMatrix(input);
 
-            Assert.AreEqual(expectedResult,solution);
+            //Assert.AreEqual(expectedResult,solution);
         }

```

Comment:
This test function does not test anything. Please verify the output of your method (indices of the array which should be 0) by using something like 
"Assert.AreEqual(0, solution[0, 1])"
"Assert.AreEqual(0, solution[2, 0])"

Reply:
what is the input array here?
maybe you can define an array like following

int[,] inputArray= new int[3, 4]  { { 1, 1, 1, 1 },
                                                    { 2, 2, 2, 2 },
                                                    { 3, 3, 0, 3}, };

and then you can make assertions(check also other indexes to make sure that you didn't convert them to 0 ). 

Also do not forget to put the "[TestMethod]" decorator before your test method in order for your test framework to understand that this function is a test method.

Reply:
i define an array and test functions . Look here e2fc83e7517147dca2cb5d463eb1c14b589da350

Reply:
Please change "[DataTestMethod]" to just "[TestMethod]" since it's not a parametric test.
Also, could ye check other indexes to make sure that you didn't convert them to 0. like as 

Assert.AreEqual(1, solution[0, 0]); 
Assert.AreEqual(2, solution[1, 3]); 


Write another unit test for following example:

int[,] matrix = new int[3, 3] { { 1, 1, 1 },
                                            { 2, 0, 2 },
                                            { 3, 3, 3 } };

Reply:
I did what you say. Look here f0c5df5bd50b86f2a9b2fe216d0ced5a8de70618

Reply:
Do not comment out the previous matrix, I meant for you to write a unit test with a different matrix. So in the end, you will have two unit tests for your method, one of them will have input as 
 { { 1, 1, 1, 1 },
   { 2, 2, 2, 2 },
   { 3, 3, 0, 3}, };

and the other one will have the 
{ {1,2,3 },
  {1,0,1 },
  {1,1,1} }

.So basically, copy your test and create a new one, just modify the input part and assertions.

Reply:
I did what you say. Look here 238357bbcf951b68c8f9faf0a338945823858e28 but i commited wrong file :)

Reply:
What I ment was creating two different tests for each input. I added one commit to your pr -> d4e8d5db2bd8cd94de284e23c40a5c73b63ee1a5